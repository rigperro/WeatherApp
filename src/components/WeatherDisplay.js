import React from 'react';

const WeatherDisplay = ({city, temp, error}) => {
  return(
    <div>
    {city ?
      <>
      <p>Current weather for: <b>{city}</b></p>
      <p>Temperature: <b>{temp} Â°C</b></p>
      </>
    :
      <>
        <p> To use this app, type the name of a city in the field below
        and press Enter or Go! </p>
        {<p>{ error }</p>}
      </>
      }
    </div>
  );
};

export default WeatherDisplay;
