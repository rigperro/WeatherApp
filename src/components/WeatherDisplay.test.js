import React from 'react';
import ReactDOM from 'react-dom';
import WeatherDisplay from './WeatherDisplay';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

it('renders without chrashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WeatherDisplay />, div);
});

test('WeatherDisplay renders correctly', () => {
  const tree = renderer
    .create(<WeatherDisplay city="Helsinki" temp="0" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
