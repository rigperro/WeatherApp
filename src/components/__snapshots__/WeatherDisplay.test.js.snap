// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDisplay renders correctly 1`] = `
<div>
  <p>
    Current weather for: 
    <b>
      Helsinki
    </b>
  </p>
  <p>
    Temperature: 
    <b>
      0
       Â°C
    </b>
  </p>
</div>
`;
